include/seqan3/range/view/pairwise_combine.hpp

```diff
@@ -0,0 +1,595 @@
+template <std::ranges::Range range_type>
+    requires !std::ranges::View<range_type>
+pairwise_combine_view(range_type & range) -> pairwise_combine_view<decltype(std::declval<range_type &>() | view::all)>;
+//!\}
+
+} // namespace seqan3::detail
+
+namespace seqan3::view
+{
+/*!\name General purpose views
+ * \{
+ */
+
+/*!\brief             A view adapter that generates all pairwise combinations of the elements of the source range.
+ * \tparam urng_t     The type of the range being processed. See below for requirements.
+ * \param[in] urange  The range being processed.
+ * \returns           A range over all pairwise combinations. See below for the properties of the returned range.
+ * \ingroup view
+ *
+ * \details
+ *
+ * This view generates two-element tuples representing all possible combinations of the elements of the source range
+ * while ignoring the order of the elements. If the source range has less than two elements the returned range is empty,
+ * otherwise the size of the returned view corresponds to the binomial coefficient `n choose 2`, where `n` is the
+ * size of the source range. The reference type of this range is a tuple over the reference type of the source range.
+ * Constness will not be propagated to the reference types of the source range.
+ *
+ * ### View properties
+ *
+ * | range concepts and reference_t  | `urng_t` (underlying range type)      | `rrng_t` (returned range type)                                       |
+ * |---------------------------------|:-------------------------------------:|:--------------------------------------------------------------------:|
+ * | std::ranges::InputRange         |                                       | *undefined*                                                          |
+ * | std::ranges::ForwardRange       | *required*                            | *preserved*                                                          |
+ * | std::ranges::BidirectionalRange |                                       | *preserved*                                                          |
+ * | std::ranges::RandomAccessRange  |                                       | *preserved*                                                          |
+ * | std::ranges::ContiguousRange    |                                       | *lost*                                                               |
+ * |                                 |                                       |                                                                      |
+ * | std::ranges::ViewableRange      | *required*                            | *guaranteed*                                                         |
+ * | std::ranges::View               |                                       | *guaranteed*                                                         |
+ * | std::ranges::SizedRange         |                                       | *guaranteed* iff urng_t models std::RandomAccessRange                |
```

Comment:
This is not required, but you guarantee it? How? What about an infinite random access range as underlying type?

Reply:
How about distance of an infinite random access range?

Reply:
ja, ja, ja, ... I know it does not mean that it is sized.
But honestly, this view won't work with infinity ranges anyway, because upon construction it will take forever to get to the end - 1 position unless it was made finite before.

Reply:
I added a block to make clear that it won't work with infinite ranges .

Reply:
Out of curiosity: in theory it could work for non-common infinite random access ranges, though, or not? It would just creates infinit pairs with the first element, though...

In practice: Is there a way to check from the code and throw an exception on construction? Infinite run-time loop is quite hard to catch. Then again if you formally require CommonRange, we exclude inifinite ranges anyway, or not? 

Actually no. We could have a common forward view where end is just some iterator of the same type that always compares false thus modelling an infinite common range... 🤔

Reply:
>In practice: Is there a way to check from the code and throw an exception on construction? Infinite run-time loop is quite hard to catch. Then again if you formally require CommonRange, we exclude inifinite ranges anyway, or not?
Actually no. We could have a common forward view where end is just some iterator of the same type that always compares false thus modelling an infinite common range... 🤔

Exactly. CommonRange just wraps the iterator and the sentinel but resolves to the comparison of the wrapped objects. Thus, an infinite range can be a common range. I could refactor the code, such that I don't need to scan for the last element before, but only for the sake of efficiency for the increment and decrement operators. Though, the runtime should be compared before making premature assumptions. But since it does not break API I would postpone this.

Reply:
I did some reading and we can at least assume that a bidirectional common iterator is not infinite, because decrementing the end would be undefined. For this reason `view::common` also downgrades everything to at most ForwardRange.

Reply:
> I did some reading and we can at least assume that a bidirectional common iterator is not infinite, because decrementing the end would be undefined. For this reason view::common also downgrades everything to at most ForwardRange.

Well, all natively common iterators would work. The problem that I wasn't aware of really is that many views are not common and that common properties are really just a way of providing access to legacy code but very limited as it mostly downgrades to forward ranges. So by the way of how views work they should in general do not require common ranges. But also note, that this little difference will touch probably only a very few use cases.